File: DTC.Api/Dtos/Board/BoardMatchDto.cs
using DTC.Api.Dtos.MatchParticipant;
using DTC.Api.Enums;

namespace DTC.Api.Dtos.Board
{
    public class BoardMatchDto
    {
        public int MatchId { get; set; }
        public DateTimeOffset? PlannedStart { get; set; }
        public DateTimeOffset? ActualStart { get; set; }
        public MatchStatus Status { get; set; }
        public List<MatchParticipantDto> Participants { get; set; } = new();
    }
}

File: dtc.frontend/src/dtos/board/BoardMatchDto.ts
import z from "zod";
import { matchParticipantDtoSchema } from "../MatchParticipant/MatchParticipantDto";
import { MatchStatus } from "../../enums/MatchStatus";

export const boardMatchDtoSchema = z.object({
  matchId: z.number(),
  roundId: z.number(),
  roundName: z.string().nullable(),
  tournamentName: z.string(),
  plannedStart: z.iso.datetime({ offset: true }).nullable(),
  actualStart: z.iso.datetime({ offset: true }).nullable(),
  status: z.enum(MatchStatus),
  participants: z.array(matchParticipantDtoSchema),
});

export type BoardMatchDto = z.infer<typeof boardMatchDtoSchema>;
